pkg/tcpip/stack/stack.go

```diff
@@ -578,6 +579,9 @@ func (s *Stack) SetRouteTable(table []tcpip.Route) {
 	s.route.mu.Lock()
 	defer s.route.mu.Unlock()
 	s.route.mu.table = table
+	sort.Slice(s.route.mu.table, func(i, j int) bool {
+		return s.route.mu.table[i].Destination.Prefix() < s.route.mu.table[j].Destination.Prefix()
+	})
```

Comment:
I'm not sure that sorting is what we want to do here in every case? maybe the table should be pre-sorted if we care about that? I know netstack tries not to add too much Linux specific behavior and I'm not sure if this qualifies.

Probably not that big of a deal but other folks like @hbhasker could say for sure.

Reply:
It is very important that the table traversal would be from most specific to least specific. Currently, it traverse sequentially. The Linux dump table does not necessarily provide the entries in this order.

Reply:
Yeah, I get that. I'm 90% sure it's ok, but just 100% sure it should be done here. I'll ask for more input.

Reply:
Note that I also reversed the order of evaluation.

Reply:
After some discussion on our end, I think we want to allow callers to specify the route order so we'll sort on the caller and have routes be specified in evaluation order to the stack.

Given that, I think we want to change your AddRoute method on the netstack Stack to a signature that looks something like this:
func (s *Stack) ModifyRouteTable(func(routeTable []tcpip.Route) []tcpip.Route)

that method will accept a function that returns new route table and the caller will maintain the route sorting in the closure. That way the stack can allow modification by the caller but be able to use s.route.mu to support concurrent modification. Does that make sense?

Reply:
This signature is impossible, as the Stack interface is implemented by `host` and `inet`, and so it cannot receive `tcpip.Route`.

Reply:
Moreover, the `s.route.mu` is inaccessible outside.

Reply:
Apologies for the confusion. I meant to say that `ModifyRouteTable` should be on `tcpip.Stack`.

I'm suggesting that `AddRoute` on `inet.Stack` can probably stay as you have written it. However, the `pkg/sentry/socket/netstack` implementation of `AddRoute` will call the `ModifyRouteTable` signature I describe above in `tcpip.Stack` and maintain sorting in `pkg/sentry/socket/netstack`. `ModifyRouteTable` will be on `tcpip.Stack` and have access to `m.route.mu`.